File: lua/game.lua
local GrpcServer = require("grpc_server")

local Game = {}

function sleep(n)
    local t0 = os.clock()
    while os.clock() - t0 <= n do end
end

function Game:run()
    local server = GrpcServer:new()
    server.game = self
    self.num = 1
    self.tbl = {}
    self.player = {}
    local i = 0
    while true do
        i = i+1
    --    print ('gaming' .. i)
        server:handle_request()
        sleep(0.015)
    end
end

function Game:foo()
    print(self)
    print 'foo'
    -- error 'foo'
    return 'foo'
end

Game:run()

File: game.py
# !python2
# -*- coding: utf-8 -*-

from grpc_server import GrpcServer


class Player(object):
    pass

class Game(object):
    def __init__(self):
        self.num = 2
        self.dct = {}
        self.lst = []
        self.player = Player()

    def foo(self):
        print 'foo'
        return 'foo'

    def pnt(self, arg):
        print arg

    def run(self):
        server = GrpcServer()
        server.game = self
        server.start()
        try:
            while True:
                server.handle_request()
        except KeyboardInterrupt:
            server.shutdown()


if __name__ == '__main__':
    game = Game()
    game.run()
